brave/brave-core
# [iOS] Bitcoin account creation

Resolves https://github.com/brave/brave-browser/issues/36811



## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues/36811) for my issue
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
In order to test this PR, you need to set up a feature flag in two ways:
1. If you are building this PR locally on your Xcode, duplicate `Debug` scheme and add `--enable-features=BraveWalletBitcoin` in your `Arguments Passed On Launch`, then build.
2. If you are testing using a TF build. Add `BraveWalletBitcoin` under Settings/BraveCore Switches/Enable Features (Enable `Enable Features` and a fresh launch is required)

Note: Bitcoin balance fetching will be addressed in a separate issue https://github.com/brave/brave-browser/issues/36966
All BTC balance will be displayed as 0 value. Send BTC will be addressed in a separate issue https://github.com/brave/brave-browser/issues/36968

**Test BTC default user asset:**
1. Fresh install Brave
2. Create a new wallet 
3. Observe BTC on Mainnet is displayed in Portfolio 
4. Go to BTC on mainnet asset details 
5. Observe BTC price and price history is rendered
6. Go to wallet setting to turn on test networks 
7. Select Bitcoin Testnet in filter and save
8. Observe BTC on Testnet is displayed in Portfolio 
9. 4. Go to BTC on mainnet asset details 
5. Observe BTC price and price history is rendered

**Test Bitcoin Account creation**
1. Go to BTC on mainnet asset details 
2. Tap Send 
3. You will be prompt with an alert since there is no Bitcoin Account yet 
4. Tap yes to create a new bitcoin account 
5. Observe `Bitcoin Mainnet` is pre-selected and cannot be changed
6. Observe there is no import private key section 
7. Input an account name and tap `Add`
8. Account creation screen will be dismissed, and you will be redirect to send screen.
9. Go to Account Tab
10. Observe the newly created bitcoin account is displayed 
11. Tap the bitcoin account to go to the account details
12. Observe there is no address displayed in the screen
13. Tap `Asset`
14. Check if the visible asset is displayed correctly. 
15. Go back to Account Tab and create a new bitcoin account on testnet via `...` on top right in navigation bar
16. Choose Bitcoin Testnet and input an account name, then tap `Add`
17. Observe a new Bitcoin account on testnet has been created and displayed 
18. repeat step 11 to 14 to make sure account and visible asset are being displayed correctly. 

**Test Bitcoin Account Update**
1. Either tap `...` on Bitcoin Account Card then choose `Edit` in the context menu or go to account details. Tap `...` on top right in navigation bar and choose `Edit` in the context menu
2. It will open a new screen and only account name can be edited 
3. Edit the name and tap `Done`
4. Observe the account name has been updated through out the wallet. 

**Test BTC or Bitcoin Account Deposit**
Flow is the same 
1. Check deposit address and its QR code are being displayed correctly. 
2. Use this deposit address and faucet some BTC, after faucet is finished. This address should get updated to another one. 

**Check there is no import or export available for bitcoin accounts.**
